flexmeasures/data/schemas/scheduling/__init__.py

```diff
@@ -260,6 +267,51 @@ def forbid_fixed_prices(self, data: dict, **kwargs):
                 "Fixed prices are not currently supported for production_price in flex-context fields in the DB."
             )
 
+    @validates_schema
+    def validate_fields_unit(self, data: dict, **kwargs):
+        """Check that each field value has a valid unit."""
+
+        self._validate_price_fields(data)
+        self._validate_power_fields(data)
+        self._validate_inflexible_device_sensors(data)
+
+    def _validate_price_fields(self, data: dict):
+        """Validate price fields."""
+        self._validate_fields(data, "price", is_energy_price_unit)
+
+    def _validate_power_fields(self, data: dict):
+        """Validate power fields."""
+        self._validate_fields(data, "power", is_power_unit)
+
+    def _validate_fields(self, data: dict, field_type: str, unit_validator):
+        """Validate fields based on type and unit validator."""
+        fields = [field for field in data if field_type in field]
```

Comment:
I would go for a more explicit mapping of field name to the unit validator to be used.

Here, you check if the word "power" or "price" is in the name of the field. That is a bit too implicit for me. Capacity fields are checked in some other way, not sure where, but `validate_fields_unit()` should cover all fields. Just checking in the name might also miss future fields we add later.

So we can have a dict, mapping each field to its unit validator (this is nice to digest for every reader), and a simple loop checking each field with `_validate_fields()` (which in turn can also be a bit simpler).

Reply:
Also fine :)